### ISSUE TITLE
Update AKS cluster fails

### ISSUE THREAD

**Issue by davidobrien1985** [pulumi/pulumi-azure#212]
Deployed an Azure AKS cluster in version `1.11.9` with the intent of testing the upgrade to `1.12.7` through Pulumi.

`pulumi version` : `0.17.5`
`pulumi_azure`: `0.17.4`
Runtime: python

```
pulumi:pulumi:Stack (pulumi-azure-aks-dev):
    panic: interface conversion: interface {} is nil, not map[string]interface {}
    goroutine 89 [running]:
    github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm.expandKubernetesClusterAddonProfiles(0xc0012c4d90, 0xc000cbdc80)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_kubernetes_cluster.go:752 +0x93f
    github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmKubernetesClusterCreateUpdate(0xc0012c4d90, 0x284d760, 0xc000b52000, 0x24, 0x46e6000)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_kubernetes_cluster.go:538 +0x41c
    github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc000022b60, 0xc000b79040, 0xc000aeff80, 0x284d760, 0xc000b52000, 0xc000aeff01, 0x0, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:231 +0x250
    github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc0006d2620, 0xc00104b820, 0xc000b79040, 0xc000aeff80, 0xc000aeff80, 0x0, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:283 +0x9c
    github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi-terraform/pkg/tfbridge.(*Provider).Update(0xc0004fa500, 0x2c58620, 0xc000b161b0, 0xc000b78e60, 0xc0004fa500, 0x1, 0x1)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi-terraform/pkg/tfbridge/provider.go:720 +0x776
    github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Update_Handler.func1(0x2c58620, 0xc000b161b0, 0x278d100, 0xc000b78e60, 0x2c58620, 0xc000b161b0, 0x2c73a60, 0x46e45f8)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1367 +0x86
    github.com/pulumi/pulumi-azure/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x2c58620, 0xc000b161b0, 0x278d100, 0xc000b78e60, 0xc000aee540, 0xc000aee560, 0x0, 0x0, 0x2c42f60, 0xc0002eab90)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:57 +0x2b0
    github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go._ResourceProvider_Update_Handler(0x27ed580, 0xc0004fa500, 0x2c58620, 0xc000caaa20, 0xc000b78e10, 0xc0002dac40, 0x0, 0x0, 0xc001036000, 0x5678)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/github.com/pulumi/pulumi/sdk/proto/go/provider.pb.go:1369 +0x158
    github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc000574f00, 0x2c70440, 0xc0002e8a80, 0xc0004fb500, 0xc0006fa2a0, 0x46b8000, 0x0, 0x0, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:966 +0x4a2
    github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).handleStream(0xc000574f00, 0x2c70440, 0xc0002e8a80, 0xc0004fb500, 0x0)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:1245 +0xd58
    github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0003bc020, 0xc000574f00, 0x2c70440, 0xc0002e8a80, 0xc0004fb500)
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:685 +0x9f
    created by github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
    	/home/travis/gopath/src/github.com/pulumi/pulumi-azure/vendor/google.golang.org/grpc/server.go:683 +0xa1
 
    error: update failed
```

**Comment by lukehoban** [comment]
This looks like the same issue as https://github.com/pulumi/pulumi/issues/2511. 

However, the fact that this is getting triggered by trivial updates like just bumping the version suggests this is more serious.

**Comment by davidobrien1985** [comment]
Seems similar, yes.
The upgrade works via the `az` cli and I know I've done this in terraform recently.

**Comment by jen20** [comment]
This will be fixed by #245. Thanks for reporting, @davidobrien1985!

**Comment by stack72** [comment]
Hi @davidobrien1985 	

Please can you tell me if you had a chance to try the fix that was merged in #245?

We believe this will solve your issues

Thanks

Paul

**Comment by stack72** [comment]
Hi @davidobrien1985 	

I am going to go ahead and close this out. I am hoping that your issue is resolved, but if there is still a problem for you, then please do let me know

Paul
